geoh5py/shared/merging/base.py

```diff
@@ -91,12 +91,17 @@ def merge_data(
                         if association == "VERTEX"
                         else out_entity.n_cells
                     )
+
+                    # update the entity type
+                    entity_type = data.entity_type
+                    entity_type.workspace = out_entity.workspace
```

Comment:
We can't do that as it will change the workspace of the original data. 
Consider making a copy the entity_type in the new workspace instead.

Reply:
Refer to my first answer: what about creating a copy function then?

Reply:
Agree, a copy method on entity_type would work.